doc/developer/design/20240411_graceful_reconfiguration_of_managed_clusters_v1.md

```diff
@@ -0,0 +1,367 @@
+some user defined duraiton while the new replicas spin up. This mechanism should
+allow for a follow on feature, where teardown is delayed until the newly created
+replicas have finished hydrating, or reach some parity with the replicas they
+are replacing.
+
+## V1 Scope
+ - Timeout based reconfiguration cleanup scheduling
+
+## V2 Scope
+ - Hydration based reconfiguration cleanup scheduling
+
+# Out of scope
+ - Issues arising from multi-subscriber message handling.
+
+## Solution Proposal
+
+#### Summary:
+This feature will introduce new SQL in `ALTER CLUSTER`
+which will delay the cleanup of replicas until a specified timeout, or
+until a hydration check is triggered. It will be possible to preempt the cleanup
+by respecifying the statement with a TIMEOUT of 0.
+Ex:
+```sql
+ALTER CLUSTER c1 SET (SIZE 'small') WITH ( CLEANUP [AUTO DETECT] TIMEOUT 5 minutes );
+```
+Here, `AUTO DETECT` indicates we would automatically detect via some parity check that the
+new replicas are as hydrated as the prior replicas. For now a timeout must be specified.
+
+The new syntax and mechanisms will be built to work with auto cleanup mechanisms
+built around cluster hydration but the syntax for that and the mechanism are not
+in scope for this document.
+
+All replicas will be billed during reconfiguraiton (except explicitly unbilled replicas).
+
+
+#### Definitions:
+  - Active Replica: These replicas should be actively serving results and are
+  named r1,r2,r3,etc... if a reconfiguration is ongoing they will match the
+  prior version of the configuration. If a reconfiguration is not ongoing they
+  should be the only replicas.
+  - Reconfiguration Replica: These replicas only exist during a
+  reconfiguration , their names will be suffixed with `reconfig-`. They will
+  move to active once the reconfiguration is finalized.
+
+
+#### Guard Rails
+- Only one reconfigure can occur at a time for a given cluster. If a cluster
+  undergoing reconfiguration is modified we should handle that in a way that
+  supplants the ongoing reconfiguratin with a new reconfiguration, or cancels
+  reconfiguration all together.
+- Protect against creating a source on a reconfiguring cluster, and
+  reconfiguring a cluster with a source.. these clusters will need to use the
+  non schedule mechanism (delete-before-create).
+
+````
+Possible guard rail
+- Limiting the timeout Duration. In theory one could set an arbitrarily long
+  duration, it maybe be reasonable to introduce some maximum to the timeout
+  duration.
+````
+
+#### Visisibilty
+Reconfiguration replicas should be identifiable for purposes of reporting and visibility
+in the UI. We may be able to do this by create a prefix "reconfig-" for
+reconfiguration replicas, alternatively we could set a bool status on the
+replica resource in the catalog.
+
+### Details
+
+#### Invariants:
+  - Only one reconfiguration will be allowed at a time for a cluster.
```

Comment:
yeah, I'm not really sure what the implications are of switching the cluster_schedule from a single value to a vec. 
Having two refresh schedules doesn't make much sense nor does having two reconfigure schedules. Perhaps it's more clear to say that the list of schedules must be unique with respect to schedule type and for reconfigure this will be resolved by overwriting the existing reconfigure.